core/src/main/java/org/apache/iceberg/actions/rewrite/BinPackStrategy.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions.rewrite;
```

Comment:
I am not sure `rewrite` package is very descriptive. I'd either put directly into `actions` as we don't have separate packages for other actions or try to come up with a more descriptive name as we have rewrite manifests, rewrite data files.

Reply:
sure, I think it's fine to keep them in the same package if you like, I think it just kept things a little neater since these aren't actions themselves

Reply:
+1 for placing it in `actions`, currently it only has the action and result classes, but I think it is reasonable enough to have this under that path.

Reply:
Actually, I did not notice `org.apache.iceberg.actions.rewrite` is used as the package path for `RewriteStrategy` in the API module. We should probably keep those things consistent, if we think placing `BinPackStrategy` in `actions` is preferred, then I think we should also change the path for the API.

Reply:
Yep moving both files

Reply:
This has been fixed.